src/plugins/signoff/index.js

```diff
@@ -52,20 +84,59 @@ class SignoffButton extends React.Component {
       return null;
     }
 
+    const {status="work-in-progress"} = collectionState.data;
+    let action;
+    let label;
+    if (status === "to-review") {
+      return (
+        <span>
+          <a className="btn btn-success"
+             href="#"
+             onClick={(event) => {
+               event.preventDefault();
+               dispatch(approveChanges());
+             }}><i className="glyphicon glyphicon-ok"></i> Approve changes</a>
+          <a className="btn btn-danger"
+             href="#"
+             onClick={(event) => {
+               event.preventDefault();
+               dispatch(declineChanges());
+             }}><i className="glyphicon glyphicon-remove"></i> Decline changes</a>
+
+        </span>
+      );
+    } else if (status === "signed") {
+      action = approveChanges();
+      label = "Re-sign";
+      return (
+          <a className="btn btn-info"
+             href="#"
+             onClick={(event) => {
+               event.preventDefault();
+               dispatch(action);
+             }}>{label}</a>
+      );
+    }
+
+    // Default to request review
+    action = requestReview();
+    label = "Request review";
     return (
-      <a className="btn btn-info"
-         href="#"
-         onClick={(event) => {
-           event.preventDefault();
-           dispatch(requestSignoff());
-         }}>Request signoff</a>
-    );
+        <a className="btn btn-info"
+           href="#"
+           onClick={(event) => {
+             event.preventDefault();
+             dispatch(action);
+           }}>{label}</a>
+      );
   }
 }
 
 
 export const sagas = [
-  [takeEvery, PLUGIN_SIGNOFF_REQUEST, handleSignoffRequest]
```

Comment:
I see the interest in using a single saga for all the events, but I wonder if we shouldn't have three sagas leveraging a single function to handle common stuff?